xml_converter/src/xml_converter.cpp

```diff
@@ -32,47 +33,121 @@
 #include "trail_gen.hpp"
 
 using namespace std;
+using namespace Attribute::UniqueId;
 
-map<string, vector<string>> read_taco_directory(
+struct CategoryWithinSinglePack {
+    Category* category;
+    vector<string> file_paths;
+};
+
+struct CategoryWithinMultiplePacks {
+    vector<CategoryWithinSinglePack> categories;
+};
+
+map<UniqueId, CategoryWithinSinglePack> read_taco_directory(
     string input_path,
     map<string, Category>* marker_categories,
     vector<Parseable*>* parsed_pois
 ) {
-    map<string, vector<string>> top_level_category_file_locations;
+    map<UniqueId, CategoryWithinSinglePack> top_level_category_file_locations;
     if (!filesystem::exists(input_path)) {
         cout << "Error: " << input_path << " is not an existing directory or file" << endl;
     }
-
+    map<UniqueId, map<string, vector<string>>> id_conflicts;
     string directory_name = filesystem::path(input_path).filename();
     vector<MarkerPackFile> xml_files = get_files_by_suffix(input_path, ".xml");
     for (const MarkerPackFile& path : xml_files) {
-        set<string> top_level_category_names = parse_xml_file(path, marker_categories, parsed_pois);
+        map<UniqueId, Category*> top_level_categories = parse_xml_file(path, marker_categories, parsed_pois);
         string file_path = join_file_paths(input_path, path.relative_filepath);
-        for (set<string>::iterator it = top_level_category_names.begin(); it != top_level_category_names.end(); it++) {
-            top_level_category_file_locations[*it].push_back(file_path);
+        for (map<UniqueId, Category*>::iterator it = top_level_categories.begin(); it != top_level_categories.end(); it++) {
+            if (top_level_category_file_locations.find(it->first) != top_level_category_file_locations.end()) {
+                if (top_level_category_file_locations[it->first].category != it->second) {
+                    id_conflicts[it->first][top_level_category_file_locations[it->first].category->name] = top_level_category_file_locations[it->first].file_paths;
+                    id_conflicts[it->first][it->second->name].push_back(file_path);
+                    continue;
+                }
+            }
+            else {
+                top_level_category_file_locations[it->first].category = it->second;
+            }
+            top_level_category_file_locations[it->first].file_paths.push_back(file_path);
```

Comment:
I think this logic is a little convoluted and needs to be simplified as we discussed offline.